global-news-app/src/api/util.ts

```diff
@@ -0,0 +1,91 @@
+import { format } from "date-fns";
+import { Selections } from "Types";
+import { SelectionsKeys } from "Types/enums";
+import { baseURL } from "./client";
+
+export const buildUrl = (
+  selections: Selections,
+  userLocation: string = "",
+  pageParam?: string,
+  pageSize?: string
+): string => {
+  const {
+    articleType,
+    dates,
+    country,
+    sources,
+    category,
+    q,
+    sortBy,
+    language,
+  } = selections;
+
+  const formattedDates = formattedDatesApi(dates!);
+  const isTopHeadlines = articleType?.value === SelectionsKeys.TOP_HEADLINES;
+
+  const formatteAarticleType = isTopHeadlines
+    ? SelectionsKeys.TOP_HEADLINES
+    : SelectionsKeys.EVERYTHING;
+
+  const url = new URL(`${baseURL}${formatteAarticleType}?`);
+
+  if (!isTopHeadlines) {
+    if (sortBy) {
+      url.searchParams.append(SelectionsKeys.SORT_BY, sortBy?.value);
+    }
+    if (dates) {
+      url.searchParams.append(SelectionsKeys.FROM, formattedDates?.from ?? "");
+      url.searchParams.append(SelectionsKeys.TO, formattedDates?.to ?? "");
+    }
+    if (language) {
+      url.searchParams.append(SelectionsKeys.LANGUAGE, language.value);
+    }
+  } else if (isTopHeadlines) {
+    if (country) {
+      url.searchParams.append(SelectionsKeys.COUNTRY, country?.value);
+    }
+    if (category) {
+      url.searchParams.append(SelectionsKeys.CATEGORY, category.value);
+    }
+  }
+
+  if (sources) {
+    url.searchParams.append(SelectionsKeys.SOURCES, sources.value);
+  }
+  if (q) {
+    url.searchParams.append("q", q);
+  }
+  if (pageParam) {
+    url.searchParams.append("page", pageParam);
+  }
+  url.searchParams.append("pageSize", pageSize ?? "100");
+  url.searchParams.append("apiKey", API_KEY);
+
+  return url.toString();
+};
+
+export const formattedDatesApi = (dates: [Date, Date]) => {
+  if (!dates?.length) return null;
+  const newdates = {
+    from: format(dates[0], "yyyy-MM-dd"),
+    to: format(dates[1], "yyyy-MM-dd"),
+  };
+  return newdates;
+};
+
+const api_keys = [
```

Comment:
api keys can be sensitive information
maybe add the to the .env file and the load them to an array in case that you need an array